component/prometheus/remotewrite/types.go

```diff
@@ -41,45 +51,109 @@ var (
-	WALOptions     WALOptions        `river:"wal,block,optional"`
+// Arguments represents the input state of the prometheus.remote_write
+// component.
+type Arguments struct {
+	ExternalLabels map[string]string  `river:"external_labels,attr,optional"`
+	Endpoints      []*EndpointOptions `river:"endpoint,block,optional"`
+	WALOptions     WALOptions         `river:"wal,block,optional"`
 }
 
 // UnmarshalRiver implements river.Unmarshaler.
-func (rc *RemoteConfig) UnmarshalRiver(f func(interface{}) error) error {
-	*rc = DefaultRemoteConfig
+func (rc *Arguments) UnmarshalRiver(f func(interface{}) error) error {
+	*rc = DefaultArguments
 
-	type config RemoteConfig
+	type config Arguments
 	return f((*config)(rc))
 }
 
-// Config is the metrics_fowarder's configuration for where to send
-// metrics stored in the WAL.
-type Config struct {
+// EndpointOptions describes an individual location for where metrics in the WAL
+// should be delivered to using the remote_write protocol.
+type EndpointOptions struct {
 	Name             string                  `river:"name,attr,optional"`
 	URL              string                  `river:"url,attr"`
+	RemoteTimeout    time.Duration           `river:"remote_timeout,attr,optional"`
+	Headers          map[string]string       `river:"headers,attr,optional"`
 	SendExemplars    bool                    `river:"send_exemplars,attr,optional"`
-	BasicAuth        *BasicAuthConfig        `river:"basic_auth,block,optional"`
-	QueueConfig      *QueueConfig            `river:"queue_config,block,optional"`
-	HTTPClientConfig *types.HTTPClientConfig `river:"client_config,block,optional"`
+	HTTPClientConfig *types.HTTPClientConfig `river:"http_client_config,block,optional"`
+	QueueOptions     *QueueOptions           `river:"queue_config,block,optional"`
+	MetadataOptions  *MetadataOptions        `river:"metadata_config,block,optional"`
+}
+
+// UnmarshalRiver implements river.Unmarshaler.
+func (r *EndpointOptions) UnmarshalRiver(f func(v interface{}) error) error {
+	*r = DefaultEndpointOptions
+
+	type arguments EndpointOptions
+	return f((*arguments)(r))
 }
 
-// QueueConfig handles the low level queue config options for a remote_write
-type QueueConfig struct {
+// QueueOptions handles the low level queue config options for a remote_write
+type QueueOptions struct {
 	Capacity          int           `river:"capacity,attr,optional"`
 	MaxShards         int           `river:"max_shards,attr,optional"`
 	MinShards         int           `river:"min_shards,attr,optional"`
 	MaxSamplesPerSend int           `river:"max_samples_per_send,attr,optional"`
 	BatchSendDeadline time.Duration `river:"batch_send_deadline,attr,optional"`
 	MinBackoff        time.Duration `river:"min_backoff,attr,optional"`
 	MaxBackoff        time.Duration `river:"max_backoff,attr,optional"`
-	RetryOn429        bool          `river:"retry_on_http_429,attr,optional"`
+	RetryOnRateLimit  bool          `river:"retry_on_http_429,attr,optional"`
```

Comment:
Should we keep the 429 variable name so it aligns more closely? Or my preference would be to align on `retry_on_rate_limit`.

Reply:
I'll keep the variable name 👍

Reply:
Changed to RetryOnHTTP429 so the name is more clear.